src/transformers/custom-elements-manifest/get-declarations.ts

```diff
@@ -0,0 +1,334 @@
+import * as tsModule from "typescript";
+import { isSimpleType, toSimpleType } from "ts-simple-type";
+import { ComponentDeclaration } from "../../analyze/types/component-declaration";
+import { getMixinHeritageClauses, getSuperclassHeritageClause } from "../../analyze/util/component-declaration-util";
+import { findParent, getNodeName } from "../../analyze/util/ast-util";
+import { getJsDoc } from "../../analyze/util/js-doc-util";
+import { getTypeHintFromType } from "../../util/get-type-hint-from-type";
+import { AnalyzerResult } from "../../analyze/types/analyzer-result";
+import * as schema from "./schema";
+import { TransformerContext } from "../transformer-context";
+import {
+	typeToSchemaType,
+	getSummaryFromJsDoc,
+	getParameterFromJsDoc,
+	getReturnFromJsDoc,
+	getReferenceFromHeritageClause,
+	getInheritedFromReference
+} from "./utils";
+
+/**
+ * Returns variables in an analyzer result
+ * @param result
+ * @param context
+ */
+function* getExportedDeclarationsFromResult(result: AnalyzerResult, context: TransformerContext): IterableIterator<schema.Declaration> {
+	// Get all export symbols in the source file
+	const symbol = context.checker.getSymbolAtLocation(result.sourceFile);
+	if (symbol == null) {
+		return;
+	}
+
+	const exports = context.checker.getExportsOfModule(symbol);
+
+	// Convert all export variables to VariableDocs
+	for (const exp of exports) {
+		const node = exp.valueDeclaration;
+
+		switch (exp.flags) {
```

Comment:
Is `flags` a single value that can be compared like this? The name makes it seem like a bitmask that you'd have to compare with logical and.

Reply:
you're correct, its also ported code. ill update